python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/conftest.py

```diff
@@ -0,0 +1,260 @@
+    "data": {
+        "id": "connector_id",
+        "service": "15five",
+        "schema": "schema.table",
+        "paused": False,
+        "status": {
+            "tasks": [
+                {
+                    "code": "resync_table_warning",
+                    "message": "Resync Table Warning",
+                    "details": "string",
+                }
+            ],
+            "warnings": [
+                {
+                    "code": "resync_table_warning",
+                    "message": "Resync Table Warning",
+                    "details": "string",
+                }
+            ],
+            "schema_status": "ready",
+            "update_state": "delayed",
+            "setup_state": "connected",
+            "sync_state": "scheduled",
+            "is_historical_sync": False,
+            "rescheduled_for": "2024-12-01T15:43:29.013729Z",
+        },
+        "daily_sync_time": "14:00",
+        "succeeded_at": "2024-03-17T12:31:40.870504Z",
+        "sync_frequency": 1440,
+        "group_id": "group_id",
+        "connected_by": "user_id",
+        "setup_tests": [
+            {
+                "title": "Test Title",
+                "status": "PASSED",
+                "message": "Test Passed",
+                "details": "Test Details",
+            }
+        ],
+        "source_sync_details": {},
+        "service_version": 0,
+        "created_at": "2023-12-01T15:43:29.013729Z",
+        "failed_at": "2024-04-01T18:13:25.043659Z",
+        "private_link_id": "private_link_id",
+        "proxy_agent_id": "proxy_agent_id",
+        "networking_method": "Directly",
+        "connect_card": {
+            "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJkIjp7ImxvZ2luIjp0cnVlLCJ1c2VyIjoiX2FjY291bnR3b3J0aHkiLCJhY2NvdW50IjoiX21vb25iZWFtX2FjYyIsImdyb3VwIjoiX21vb25iZWFtIiwiY29ubmVjdG9yIjoiY29iYWx0X2VsZXZhdGlvbiIsIm1ldGhvZCI6IlBiZkNhcmQiLCJpZGVudGl0eSI6ZmFsc2V9LCJpYXQiOjE2Njc4MzA2MzZ9.YUMGUbzxW96xsKJLo4bTorqzx8Q19GTrUi3WFRFM8BU",
+            "uri": "https://fivetran.com/connect-card/setup?auth=eyJ0eXAiOiJKV1QiLCJh...",
+        },
+        "pause_after_trial": False,
+        "data_delay_threshold": 0,
+        "data_delay_sensitivity": "NORMAL",
+        "schedule_type": "auto",
+        "local_processing_agent_id": "local_processing_agent_id",
+        "connect_card_config": {"redirect_uri": "https://your.site/path", "hide_setup_guide": True},
+        "hybrid_deployment_agent_id": "hybrid_deployment_agent_id",
+        "config": {"api_key": "your_15five_api_key"},
+    },
+}
+
+
+@pytest.fixture(name="connector_id")
+def connector_id_fixture() -> str:
+    return "connector_id"
+
+
+@pytest.fixture(name="destination_id")
+def destination_id_fixture() -> str:
+    return "destination_id"
+
+
+@pytest.fixture(name="group_id")
+def group_id_fixture() -> str:
+    return "group_id"
+
+
+@pytest.fixture(
+    name="workspace_data_api_mocks_fn",
+)
+def workspace_data_api_mocks_fn_fixture(
```

Comment:
In general I find API mocks like this to be relatively inflexible and hard to parse. How would you feel about instead implementing a "client fake" similar to how we did for airlift and dlift? Something like this:
https://github.com/dagster-io/dagster/blob/master/examples/experimental/dagster-dlift/dagster_dlift/test/client_fake.py

I've found it to be more composable and human-readable than mocking. I won't necessarily block review on this, but every time I've based a testing strategy around mocks I've regretted it. I assume this is also somewhat inherited from the old testing strategy, but I think if we're spending time in this code we might as well improve the situation.

Reply:
See [comment](https://github.com/dagster-io/dagster/pull/25788#discussion_r1835003439) in next PR